consensus/checks.go

```diff
@@ -67,6 +67,18 @@ func (consensus *Consensus) isRightBlockNumAndViewID(recvMsg *FBFTMessage,
 	return true
 }
 
+func (consensus *Consensus) isPreviousBlockNumAndViewID(recvMsg *FBFTMessage,
+) bool {
+	if recvMsg.ViewID != consensus.GetCurBlockViewID()-1 || recvMsg.BlockNum != consensus.blockNum-1 {
```

Comment:
Previous block's viewID should be fetched from the block itself, not calculating with curBlock.viewID -1.